nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/SetHealthCheckServiceConfiguration.java

```diff
@@ -0,0 +1,402 @@
+    @Min(value = 0, message = "Hogging threads threshold is a percentage so must be greater than zero")
+    @Max(value = 100, message ="Hogging threads threshold is a percentage so must be less than 100")
+    private String hogginThreadsThreshold;
+
+    @Min(value = 1, message = "Hogging threads retry count must be 1 or more")
+    @Param(name = "hogging-threads-retry-count", optional = true)
+    private String hogginThreadsRetryCount;
+
+    // stuck threads property params:
+
+    @Param(name = "stuck-threads-threshold", optional = true)
+    @Min(value = 1, message = "Threshold length must be 1 or more")
+    private String stuckThreadsThreshold;
+
+    @Param(name = "stuck-threads-threshold-unit", optional = true,
+            acceptableValues = "DAYS,HOURS,MICROSECONDS,MILLISECONDS,MINUTES,NANOSECONDS,SECONDS")
+    private String stuckThreadsThresholdUnit;
+
+    // threshold properties params:
+
+    @Param(name = "threshold-critical", optional = true, defaultValue = HealthCheckConstants.THRESHOLD_DEFAULTVAL_CRITICAL)
+    @Min(value = 0, message = "Critical threshold is a percentage so must be greater than zero")
+    @Max(value = 100, message ="Critical threshold is a percentage so must be less than 100")
+    private String thresholdCritical;
+
+    @Param(name = "threshold-warning", optional = true, defaultValue = HealthCheckConstants.THRESHOLD_DEFAULTVAL_WARNING)
+    @Min(value = 0, message = "Warning threshold is a percentage so must be greater than zero")
+    @Max(value = 100, message ="Wanring threshold is a percentage so must be less than 100")
+    private String thresholdWarning;
+
+    @Param(name = "threshold-good", optional = true, defaultValue = HealthCheckConstants.THRESHOLD_DEFAULTVAL_GOOD)
+    @Min(value = 0, message = "Good threshold is a percentage so must be greater than zero")
+    @Max(value = 100, message ="Good threshold is a percentage so must be less than 100")
+    private String thresholdGood;
+
+    private ActionReport report;
+
+    @Override
+    public void execute(AdminCommandContext context) {
+        report = context.getActionReport();
+        if (report.getExtraProperties() == null) {
+            report.setExtraProperties(new Properties());
+        }
+
+        targetConfig = targetUtil.getConfig(target);
+
+        serviceType = parseServiceType(serviceName);
+
+        if (serviceType == null) {
+            String values = Arrays.asList(CheckerType.values())
+                    .stream().map(t -> t.name().toLowerCase().replace('_', '-')).collect(Collectors.joining(", "));
+            report.setMessage("No such service: " + serviceName + ".\nChoose one of: " + values
+                    + ".\nThe name can also be given in short form consisting only of the first letters of each word.");
+            report.setActionExitCode(ActionReport.ExitCode.FAILURE);
+            return;
+        }
+        serviceName = serviceType.name().toLowerCase().replace('_', '-');
```

Comment:
This could be moved into the if below, as it isn't used anywhere else

Reply:
If you refer to the `serviceName` assignment this has another function. Since the user is allowed to give the service name in various ways I updated the name here back from the type so it is printed uniformly independent of how the user was given the service name. I added a comment to make this clear.